docs/docs-beta/docs/guides/deploy/code-locations/managing-code-locations-with-definitions.md

```diff
@@ -0,0 +1,144 @@
+---
+title: "Managing code locations with Definitions"
+description: "A code location is a collection of Dagster definitions loadable and accessible by Dagster's tools. Learn to create, load, and deploy code locations."
+sidebar_position: 100
+---
+
+A code location is a collection of Dagster definitions loadable and accessible by Dagster's tools, such as the CLI, UI, and Dagster+. A code location comprises:
+
+- A reference to a Python module that has an instance of <PyObject section="definitions" module="dagster" object="Definitions" /> in a top-level variable
+- A Python environment that can successfully load that module
+
+Definitions within a code location have a common namespace and must have unique names. This allows them to be grouped and organized by code location in tools.
+
+![Code locations](/images/guides/deploy/code-locations/code-locations-diagram.png)
+
+A single deployment can have one or multiple code locations.
+
+Code locations are loaded in a different process and communicate with Dagster system processes over an RPC mechanism. This architecture provides several advantages:
+
+- When there is an update to user code, the Dagster webserver/UI can pick up the change without a restart.
+- You can use multiple code locations to organize jobs, but still work on all of your code locations using a single instance of the webserver/UI.
+- The Dagster webserver process can run in a separate Python environment from user code so job dependencies don't need to be installed into the webserver environment.
+- Each code location can be sourced from a separate Python environment, so teams can manage their dependencies (or even their Python versions) separately.
+
+## Relevant APIs
+
+| Name                              | Description                                                                                                                                       |
+| --------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------- |
+| <PyObject section="definitions" module="dagster" object="Definitions" /> | The object that contains all the definitions defined within a code location. Definitions include assets, jobs, resources, schedules, and sensors. |
+
+## Defining code locations
+
+To define a code location, create a top-level variable that contains a <PyObject section="definitions" module="dagster" object="Definitions" /> object in a Python module. For example:
+
+```python
+# definitions.py
+
+defs = Definitions(
+    assets=[dbt_customers_asset, dbt_orders_asset],
+    schedules=[bi_weekly_schedule],
+    sensors=[new_data_sensor],
+    resources=[dbt_resource],
+)
+```
+
+It is recommenced to include definitions in a Python module named `definitions.py`.
```

Comment:
@cmpadden if this is true, should Dagster U get updated and have the `Definitions` moved out of the `__init__.py`

Reply:
Yeah, that is a more recent (last year) best practice, and I think DU should be updated.